src/test-project/src/test/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixturesTest/dialogs/projectManipulation/NewProjectDialogTest.java

```diff
@@ -11,145 +11,284 @@
+        String currentlySelectedProjectSdk = TextUtils.listOfRemoteTextToString(projectJdkComboBox.findAllText());
         assertTrue(currentlySelectedProjectSdk.contains("8"), "Selected project SDK should be Java 8 but is '" + currentlySelectedProjectSdk + "'");
-        newProjectDialog.setProjectSdkIfAvailable("11");
-        currentlySelectedProjectSdk = listOfRemoteTextToString(projectJdkComboBox.findAllText());
+        newProjectDialogFirstPage.setProjectSdkIfAvailable("11");
+        currentlySelectedProjectSdk = TextUtils.listOfRemoteTextToString(projectJdkComboBox.findAllText());
         assertTrue(currentlySelectedProjectSdk.contains("11"), "Selected project SDK should be Java 11 but is '" + currentlySelectedProjectSdk + "'");
     }
 
     @Test
     public void selectNewProjectTypeTest() {
-        newProjectDialog.selectNewProjectType("Empty Project");
-        boolean isEmptyProjectLabelVisible = !newProjectDialog.findAll(JListFixture.class, byXpath("//div[@visible_text='Empty Project']")).isEmpty();
+        newProjectDialogFirstPage.selectNewProjectType("Empty Project");
+        boolean isEmptyProjectLabelVisible = !newProjectDialogFirstPage.findAll(JListFixture.class, byXpath("//div[@visible_text='Empty Project']")).isEmpty();
         assertTrue(isEmptyProjectLabelVisible, "The 'Empty Project' label should be visible but is not.");
 
-        newProjectDialog.selectNewProjectType("Java FX");
-        boolean isJavaFXApplicationLabelVisible = !newProjectDialog.findAll(JListFixture.class, byXpath("//div[@visible_text='JavaFX Application']")).isEmpty();
+        newProjectDialogFirstPage.selectNewProjectType("Java FX");
+        boolean isJavaFXApplicationLabelVisible = !newProjectDialogFirstPage.findAll(JListFixture.class, byXpath("//div[@visible_text='JavaFX Application']")).isEmpty();
         assertTrue(isJavaFXApplicationLabelVisible, "The 'Java FX' label should be visible but is not.");
     }
 
     private void navigateToSetProjectNamePage(NewProjectType newProjectType) {
-        newProjectDialog.selectNewProjectType(newProjectType.toString());
-        newProjectDialog.next();
+        newProjectDialogFirstPage.selectNewProjectType(newProjectType.toString());
+        newProjectDialogWizard.next();
         if (newProjectType == NewProjectType.PLAIN_JAVA) {
-            newProjectDialog.next();
+            newProjectDialogWizard.next();
         }
     }
 
-    private void testProjectNameInputField(String projectName, NewProjectType newProjectType) {
+    private void testProjectNameInputField(NewProjectType newProjectType) {
+        navigateToSetProjectNamePage(newProjectType);
+        AbstractFinalPage abstractFinalPage = newProjectDialogWizard.find(AbstractFinalPage.class, Duration.ofSeconds(10));
```

Comment:
This does not really look good. Are you really instantiating an abstract class? Like we know it is not abstract, but don't you have more specific data type to hold the given functionality in?

Reply:
Fixed.